app/lib/formatter.rb

```diff
@@ -288,7 +288,7 @@ def link_to_hashtag(entity)
 
   def link_html(url)
     url    = Addressable::URI.parse(url).to_s
-    prefix = url.match(/\A(https?:\/\/(www\.)?|xmpp:)/).to_s
+    prefix = url.match(/\A(https?:\/\/(www\.)?|xmpp:|gemini:\/\/)/).to_s
```

Comment:
This doesn't control what gets hyperlinked, just how URLs get visually shortened